src/shared/Core/PlatformUtils.cs

```diff
@@ -176,52 +177,109 @@ public static bool IsElevatedUser()
+                    return GetWindowsEntryPath();
                 }
 
                 if (IsMacOS())
                 {
-                    return GetMacOSArgv0();
+                    return GetMacOSEntryPath();
                 }
 
                 if (IsLinux())
                 {
-                    return GetLinuxArgv0();
+                    return GetLinuxEntryPath();
                 }
             }
             catch
             {
-                // If there are any issues getting the native argv[0]
+                // If there are any issues getting the native entry path
                 // we should not throw, and certainly not crash!
                 // Just return null instead.
             }
 
             return null;
         }
 
-        private static string GetLinuxArgv0()
+        private static string GetLinuxEntryPath()
         {
+            // Try to extract our native argv[0] from the original cmdline
             string cmdline = File.ReadAllText("/proc/self/cmdline");
-            return cmdline.Split('\0')[0];
+            string argv0 = cmdline.Split('\0')[0];
+
+            // argv[0] is an absolute file path
+            if (Path.IsPathRooted(argv0))
+            {
+                return argv0;
+            }
+
+            string path = Path.GetFullPath(
+                Path.Combine(Environment.CurrentDirectory, argv0)
+            );
+
+            // argv[0] is relative to current directory (./app) or a relative
+            // name resolved from the current directory (subdir/app).
+            // Note that we do NOT want to consider the case when it is just
+            // a simple filename (argv[0] == "app") because that would actually
+            // have been resolved from the $PATH instead (handled below)!
+            if ((argv0.StartsWith("./") || argv0.IndexOf('/') > 0) && File.Exists(path))
+            {
+                return path;
+            }
+
+            // argv[0] is a name that was resolved from the $PATH
+            string pathVar = Environment.GetEnvironmentVariable("PATH");
+            if (pathVar != null)
+            {
+                string[] paths = pathVar.Split(':');
+                foreach (string pathBase in paths)
+                {
+                    path = Path.Combine(pathBase, argv0);
+                    if (File.Exists(path))
+                    {
+                        return path;
+                    }
+                }
+            }
+
+            //
+            // We cannot determine the absolute file path from argv[0]
+            // (how we were launched), so let's now try to extract the
+            // fully resolved executable path from /proc/self/exe.
+            // Note that this means we may miss if we've been invoked
+            // via a symlink, but it's better than nothing at this point!
+            //
+            FileSystemInfo fsi = File.ResolveLinkTarget("/proc/self/exe", returnFinalTarget: false);
```

Comment:
Since this is Linux-specific code, I guess it is okay to hard-code `/proc/self/exe`. But please note that for OpenBSD, it is [`/proc/curproc/file`](https://github.com/git/git/blob/v2.38.1/config.mak.uname#L292) and for NetBSD it is [`/proc/curproc/exe`](https://github.com/git/git/blob/v2.38.1/config.mak.uname#L316) (yay standards!).

Reply:
We don't currently support BSD-variants :)

Reply:
Fair enough. I just wanted to bring it up in case GCM was ever added *BSD support.

Reply:
If we added BSD support (which, the .NET CLR does support..) there are several places we'd need to review. If/when we add support, Linux != BSD. It'll be it's own platform.